sdks/java/core/src/main/java/org/apache/beam/sdk/io/CompressedSource.java

```diff
@@ -380,18 +395,77 @@ public T getCurrent() throws NoSuchElementException {
       return readerDelegate.getCurrent();
     }
 
+    @Override
+    public final long getParallelismConsumed() {
```

Comment:
Might be easier to read if there were two versions -- a splittable implementation and a non-splittable implementation, and you used "splittable" to decide which to create.